src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -1854,51 +1721,61 @@ static MemberName generateLambdaFormInterpreterEntryPoint(MethodType mt) {
-            mv.visitInsn(Opcodes.DUP);
-            emitIconstInsn(i);
-            emitLoadInsn(basicType(ptype), i);
-            // box if primitive type
-            if (ptype.isPrimitive()) {
-                emitBoxing(Wrapper.forPrimitiveType(ptype));
+        final byte[] classFile = classFilePrologue(new Consumer<ClassBuilder>() {
+            @Override
+            public void accept(ClassBuilder clb) {
+                methodPrologue(clb, new Consumer<MethodBuilder>() {
+                    @Override
+                    public void accept(MethodBuilder mb) {
+                        mb.withFlags(ACC_STATIC);
+
+                        // Suppress this method in backtraces displayed to the user.
+                        mb.with(RuntimeVisibleAnnotationsAttribute.of(List.of(HIDDEN)));
+
+                        // Don't inline the interpreter entry.
+                        mb.with(RuntimeVisibleAnnotationsAttribute.of(List.of(DONTINLINE)));
+
+                        mb.withCode(new Consumer<CodeBuilder>() {
+                            @Override
+                            public void accept(CodeBuilder cob) {
+                                // create parameter array
+                                cob.constantInstruction(invokerType.parameterCount());
+                                cob.anewarray(CD_Object);
+
+                                // fill parameter array
+                                for (int i = 0; i < invokerType.parameterCount(); i++) {
+                                    Class<?> ptype = invokerType.parameterType(i);
+                                    cob.dup();
+                                    cob.constantInstruction(i);
+                                    emitLoadInsn(cob, basicType(ptype), i);
+                                    // box if primitive type
+                                    if (ptype.isPrimitive()) {
+                                        emitBoxing(cob, Wrapper.forPrimitiveType(ptype));
+                                    }
+                                    cob.aastore();
+                                }
+                                // invoke
+                                emitAloadInsn(cob, 0);
+                                cob.getfield(CD_MethodHandle, "form", LF);
+                                cob.swap();  // swap form and array; avoid local variable
+                                cob.invokevirtual(LF, "interpretWithArguments", MethodTypeDesc.of(CD_Object, CD_Object.arrayType()));
+
+                                // maybe unbox
+                                Class<?> rtype = invokerType.returnType();
+                                if (rtype.isPrimitive() && rtype != void.class) {
+                                    emitUnboxing(cob, Wrapper.forPrimitiveType(rtype));
+                                }
+
+                                // return statement
+                                emitReturnInsn(cob, basicType(rtype));
```

Comment:
This could avoid going through BasicType:
```suggestion
                                cob.returnInstruction(TypeKind.from(rtype));
```